pkg/apis/machine/v1beta1/machine_types.go

```diff
@@ -231,6 +238,23 @@ type MachineVersionInfo struct {
 
 /// [MachineVersionInfo]
 
+func (m *Machine) Validate() field.ErrorList {
+	errors := field.ErrorList{}
```

Comment:
Since we are adding validation as a stop gap so i am not sure it makes sense to have more robust validation like using different methods eg: `ValidateMachine()`, `ValidateMachineSpec`, `ValidateMachineMeta` , `ValidateMachineAWSProviderSpec`?

And keeping general logic like `ValidateMachine()`, `ValidateMachineSpec`, `ValidateMachineMeta` at common repo, so that it could be reused in other actuators.

Reply:
So far we validate only two bits. So, let's keep the implementation as simple as it can be. We can re-iterate on the topic once we have more to validate.